### DISCUSSION TITLE
Why is the default Gitlab scope 'api' instead of 'read_api'?

### DISCUSSION THREAD

**Discussion by haydenflinner** [woodpecker-ci/woodpecker#3298] [Ask a Question] [answered]
Note I haven't used woodpecker at all yet, still trying to set it up and ran into 'The requested scope is invalid, unknown, or malformed.' auth error when logging in with SSO due to only giving the token 'read_api' instead of 'api'. Perhaps it's for pushing git tags, merging branches and such?

Ref: https://github.com/woodpecker-ci/woodpecker/blob/da4bd8b97ddb357e11e2f18442b304242a37909b/server/forge/gitlab/gitlab.go#L44

**Comment by anbraten** [comment, marked as answer]
woodpecker is creating webhooks when enabling repos therefore it needs write access
